### PR TITLE
Troubleshooting: Add cards for product links

### PR DESCRIPTION
**Pull request by jyee27**
Add product card links for product links in a solution.
See `/Vulcan/troubleshooting_products_test` for an example wiki.

Note that neither these cards nor guide product cards support on sale formatting for prices.

<details>
<summary>Images</summary>

Desktop/tablet
--
image

Mobile
--
image

</details>

Closes: https://github.com/ifixit/ifixit/issues/47437

### CODE REVIEW

File: frontend/templates/troubleshooting/Resource.tsx [hunk 1]
```diff
@@ -1,20 +1,137 @@
...
+} from '@chakra-ui/react';
 import { Guide } from './hooks/GuideModel';
 import { FaIcon } from '@ifixit/icons';
 import { faClock } from '@fortawesome/pro-solid-svg-icons';
 import Prerendered from './prerendered';
 import { DifficultyThemeLookup, GuideDifficultyNames } from './DifficultyBadge';
+import { Product } from '@models/product';
+import { useSelectedVariant } from '@templates/product/hooks/useSelectedVariant';
+import { useIsProductForSale } from '../product/hooks/useIsProductForSale';
+import { shouldShowProductRating } from '../../../packages/helpers/product-helpers';
```
Comment (ianrohde) [inline frontend/templates/troubleshooting/Resource.tsx:25]: Is `/packages/` used enough to warrant an alias?

File: frontend/templates/troubleshooting/Resource.tsx [hunk 2]
```diff
@@ -1,20 +1,137 @@
...
          imageUrl={guide.image.thumbnail}
-         introduction={guide.introduction_rendered}
          timeRequired={guide.time_required}
          difficulty={guide.difficulty}
-      />
+         spacing="6px"
+      >
+         {guide.introduction_rendered && (
+            <Prerendered
+               lineHeight="1.36"
```
Comment (ianrohde) [inline frontend/templates/troubleshooting/Resource.tsx:41]: Was this pulled from the design? `line-height: 136%;` Just curious.

**Review by ianrohde — APPROVED** [review]
CR ⚪ 

I found an unrelated bug that aligns the images to the center of the ResourceBox Stack, but the design calls for top alignment. If we remove this line, it's fixed.

https://github.com/iFixit/react-commerce/pull/1695/files#diff-0d9fca1d316f17767a48edf73654035a0686ea985fb8028e2a1452f3e06bb0dbR200

If you're making further changes, would you mind dropping it? If not, I'll handle it later.

**Review by aburke07 — APPROVED** [review]
CR :alien:

File: frontend/templates/troubleshooting/server.tsx [hunk 3]
```diff
@@ -42,21 +43,28 @@ export const getServerSideProps: GetServerSideProps<
...
+                  },
+                  { forceMiss: hasDisableCacheGets(context) }
+               )
          )
       );
+      function isProduct(
+         product: ProductType | '' | null
+      ): product is ProductType {
+         return product !== '' && product !== null;
+      }
```
[the remark this answers is not in the document]
Comment (jyee27) [reply frontend/templates/troubleshooting/server.tsx:63]: I think I did this to fix a TypeScript error. If I can find what I looked at, I'll add a comment for it.

File: frontend/templates/troubleshooting/Resource.tsx (hunk 1 above)
Comment (jyee27) [reply frontend/templates/troubleshooting/Resource.tsx]: This should have been from `@ifixit/helpers`. Fixed in [1ee69bc](https://github.com/iFixit/react-commerce/pull/1695/commits/1ee69bce3e16567f095ee7636cfcd4fd4d3ea3c5).

File: frontend/templates/troubleshooting/server.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jyee27) [reply frontend/templates/troubleshooting/server.tsx]: [c11b404](https://github.com/iFixit/react-commerce/pull/1695/commits/c11b404585c43763e4cd4e4caa62c916a62cc2bd)

File: frontend/templates/troubleshooting/Resource.tsx (hunk 2 above)
Comment (jyee27) [reply frontend/templates/troubleshooting/Resource.tsx:40]: I think so. I didn't add this in this pull, I just moved it.

**Review by aburke07 — APPROVED** [review]
CR :alien:

**Review by ianrohde — APPROVED** [review]
CR ⚪

**Comment by jordycosta** [comment]
QA 🟢 

[Vulcan page tested](https://react-commerce-git-troubleshooting-product-cards-ifixit.vercel.app/Vulcan/troubleshooting_products_test)

Looks good on desktop + mobile ✔️ 
Clicking on the image, product title, or `Buy` button links to the product page ✔️ 

firefox_tF5yJA9L96

**Pull request merged**